dbms/src/Flash/tests/gtest_topn_executor.cpp

```diff
@@ -0,0 +1,228 @@
+
+                executeStreams(request, expect_cols[0]);
+                executeStreams(request, expect_cols[0], 2);
+                executeStreams(request, expect_cols[0], 4);
+                executeStreams(request, expect_cols[0], 8);
+            }
+        }
+    }
+
+    {
+        /// Test multi-columns
+        expect_cols = {{toNullableVec<Int32>(col_name[0], ColumnWithInt32{36, 34, 32, 27, {}, {}}),
+                        toNullableVec<String>(col_name[1], ColumnWithString{"female", "male", "male", "female", "male", "female"}),
+                        toNullableVec<String>(col_name[2], ColumnWithString{"china", "china", "usa", "usa", "china", "korea"}),
+                        toNullableVec<Int32>(col_name[3], ColumnWithInt32{900, -300, {}, 0, {}, 1300})},
+                       {toNullableVec<Int32>(col_name[0], ColumnWithInt32{32, {}, 34, 27, 36, {}}),
+                        toNullableVec<String>(col_name[1], ColumnWithString{"male", "male", "male", "female", "female", "female"}),
+                        toNullableVec<String>(col_name[2], ColumnWithString{"usa", "china", "china", "usa", "china", "korea"}),
+                        toNullableVec<Int32>(col_name[3], ColumnWithInt32{{}, {}, -300, 0, 900, 1300})},
+                       {toNullableVec<Int32>(col_name[0], ColumnWithInt32{34, {}, 32, 36, {}, 27}),
+                        toNullableVec<String>(col_name[1], ColumnWithString{"male", "male", "male", "female", "female", "female"}),
+                        toNullableVec<String>(col_name[2], ColumnWithString{"china", "china", "usa", "china", "korea", "usa"}),
+                        toNullableVec<Int32>(col_name[3], ColumnWithInt32{-300, {}, {}, 900, 1300, 0})}};
+
+        MockOrderByItems order_by_items;
+
+        {
+            /// select * from clerk order by age DESC, gender DESC;
+            order_by_items = {MockOrderByItem(col_name[0], true), MockOrderByItem(col_name[1], true)};
+            request = getRequest(table_name, order_by_items, 100);
+            executeStreams(request, expect_cols[0]);
+        }
+
+        {
+            /// select * from clerk order by gender DESC, salary ASC;
+            order_by_items = {MockOrderByItem(col_name[1], true), MockOrderByItem(col_name[3], false)};
+            request = getRequest(table_name, order_by_items, 100);
+            executeStreams(request, expect_cols[1]);
+        }
+
+        {
+            /// select * from clerk order by gender DESC, country ASC, salary DESC;
+            order_by_items = {MockOrderByItem(col_name[1], true), MockOrderByItem(col_name[2], false), MockOrderByItem(col_name[3], true)};
+            request = getRequest(table_name, order_by_items, 100);
+            executeStreams(request, expect_cols[2]);
+        }
+    }
+}
+CATCH
+
+TEST_F(ExecutorTopNTestRunner, TopNFunction)
+try
+{
+    std::shared_ptr<tipb::DAGRequest> request;
+    std::vector<ColumnsWithTypeAndName> expect_cols;
```

Comment:
Maybe not a vector?

Reply:
> Maybe not a vector?

Considering for the future tests, I think we may need vector in the future to contain more expected cols.